packages/mermaid/src/diagrams/gantt/ganttDb.js

```diff
@@ -162,6 +211,18 @@ export const getTasks = function () {
   }
 
   tasks = rawTasks;
+  if (dateRange != '') {
+    tasks = tasks.filter(function (task) {
+      let in_bounds = true;
+      if (startDateRange && task.endTime <= startDateRange) {
+        in_bounds = false;
+      }
+      if (endDateRange && task.startTime >= endDateRange) {
+        in_bounds = false;
+      }
+      return in_bounds;
```

Comment:
```suggestion
      if (startDateRange && task.endTime <= startDateRange) {
        return false;
      }
      if (endDateRange && task.startTime >= endDateRange) {
        return false;
      }
      return true;
```

You can combine the if conditions with an `||` 

```ts
if (
  (startDateRange && task.endTime <= startDateRange) ||
  (endDateRange && task.startTime >= endDateRange)
) {
  return false
}
```